src/manager.cpp

```diff
@@ -48,6 +48,9 @@ Manager::Manager(
             m_worker->getSysCfgJsonObj(), m_worker, m_ioContext);
 
 #endif
+        // set callback to detect host state change.
+        registerHostStateChangeCallback();
```

Comment:
should this not be inside the #define IBM_SYSTEM ?

Reply:
or should the performVPDRecollection(); be inside the define in the callback ? else callback is specific for recollection only.

Reply:
> should this not be inside the #define IBM_SYSTEM ?

Initially I kept it but then I thought why it should be limited to IBM_SYSTEM? If the flag is defined for a FRU in JSON then it will be called else not.
Let it be open for all.

Reply:
so you say that perform recollection is mandatory for everyone that has JSON support on a host transition

Reply:
ok acceptable